onecx/onecx-portal-ui-libs
# feat: 1-/2-button dialog component implementation

Implementation of 1-/2- button dialog component for portal-integration-angular lib.

Information:

- Component should be usable in following scenarios:
  - Directly in templates with selector and using Input/Output
  - With usage of portal-dialog.service.ts

Implementation details:
- You should be able to subscribe to dialogResult emitter while using selector or pass your own emitter as part of Input to subscribe while using portal-dialog service
- You should be able to customize dialog via "data" property passed inside DynamicDialogConfig or by passing an Input (while using selector)
- Component allows such customizations:
  - Switch between one or two buttons
  - Customize both buttons in terms of
    - icon
    - label
    - emitted value on button click
    - if button should close the dialog or not (irrelevant for selector usage)
  - Pass another component for the body of the dialog
  - Pass data for body component
- By Default Two Button dialog with Cancel/Confirm buttons should be created with a title "Title"
- By Default emitted values are false/true and dialog should be closed on click
- If you do not override default values they should be present in dialog (if you only customize main button and do not interfere with side button nor switch to one button dialog then you should get your customized button with Cancel button)

Additional infmormation:
- Tests have been written to
  - check default values persistence
  - check data passing via selector Input and DynamicDialogConfig
  - check correct usage of dialogResult
  - check if correct value is emitted
  - check if dialogRef is closed based on passed data
- Test harness for ButtonDialog has been created with ability to
  - get main button
  - get side button
  - click main button
  - click side button
- Accessibility issues found by lighthouse/axe only pointed towards p-button related implementation
